src/electos/ballotmaker/data/extractor.py

```diff
@@ -0,0 +1,311 @@
+from typing import Dict, List, Union
+
+from electos.ballotmaker.data.models import ElectionData
+from electos.datamodels.nist.indexes import ElementIndex
+from electos.datamodels.nist.models.edf import (
+    BallotMeasureContest,
+    BallotMeasureSelection,
+    BallotStyle,
+    Candidate,
+    CandidateContest,
+    CandidateSelection,
+    Election,
+    ElectionReport,
+    InternationalizedText,
+    OrderedContest,
+    OrderedHeader,
+)
+
+
+# --- Base Types
+#
+# Schema expresses these as union types not subclasses
+
+Contest = Union[BallotMeasureContest, CandidateContest]
+
+OrderedContent = Union[OrderedContest, OrderedHeader]
+
+
+# --- Utilities
+
+
+def _text_content(item):
+    """Return joined lines from internationalized text."""
+    assert isinstance(item, InternationalizedText)
+    text = "\n".join(_.content for _ in item.text)
+    return text
+
+
+def _walk_ordered_contests(content: List[OrderedContent]):
+    """Walk ordered content yielding contests."""
+    for item in content:
+        if isinstance(item, OrderedContest):
+            yield item
+        elif isinstance(item, OrderedHeader):
+            yield from _walk_ordered_contests(item.ordered_content)
+        else:
+            raise TypeError(f"Unexpected type: {type(item).__name__}")
+
+
+def _walk_ordered_headers(content: List[OrderedContent]):
+    """Walk ordered content yielding headers."""
+    for item in content:
+        if isinstance(item, OrderedHeader):
+            yield item
```

Comment:
So first off: you are right and this code needs cleaning up, or at least comments. I'll look into fixing it. But for the sake of completeness here's what's going on:

The complexity here is because `OrderedContent` and` OrderedHeader` nest inside each other. (An `OrderedHeader` is a header label and another `OrderedContent` for what's below the header.) The intent of that double `yield` is that the first gives back the label text, and the second walks the nested content.

But the reason there are two functions is because I'm not sure which of them is right -- or even if _either_ of them is. I didn't have any quick way to test them: none of our examples use `OrderedHeader`s, and I didn't want to spend time trying them out. That's the thing to do to figure it out. I've added it to my list.